### PR TITLE
Allow Switching to Git SHA or Version Tag and Restore Initial State, Improve rsync Functionality

### LINKED ISSUE #117: `instal.sh` script triggers 'rsync no such file or directory' warnings
While cleaning up errors and warnings in the installation script, I encountered a section where I have some doubts. It seems there are two rsync commands that will always execute, correct? (Lines [606](https://github.com/BU-ISCIII/relecov-platform/blob/598687f57305bbed4d11a2b4b6357c51b59df628/install.sh#L606) and [625](https://github.com/BU-ISCIII/relecov-platform/blob/598687f57305bbed4d11a2b4b6357c51b59df628/install.sh#L625)).

The second rsync command (line [625](https://github.com/BU-ISCIII/relecov-platform/blob/598687f57305bbed4d11a2b4b6357c51b59df628/install.sh#L625)) causes these warnings because it enters a directory (https://github.com/BU-ISCIII/relecov-platform/blob/598687f57305bbed4d11a2b4b6357c51b59df628/install.sh#L608) where the specified files do not exist:
```
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/README.md" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/LICENSE" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/test" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/conf" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/django_utils" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/django_plotly_dash" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/relecov_core" failed: No such file or directory (2)
rsync: [sender] link_stat "/opt/relecov-platform/relecov_platform/relecov_dashboard" failed: No such file or directory (2)
```
Additionally, the rsync command at line [625](https://github.com/BU-ISCIII/relecov-platform/blob/598687f57305bbed4d11a2b4b6357c51b59df628/install.sh#L625) looks for django_utils and django_plotly_dash, which are not present in the Git repository. Is this a typo, or are they there for a specific reason?

### PR DESCRIPTION
**Pull request by Daniel-VM**
### Fixes in this PR:
- Incorrect `cd` across directories before running rsync.
- Rename typo in file extension.
- Removes a duplicate rsync command.
- Fixes indentation issues in the script for better readability.

### Enhancements in this PR:
- Replace `--dev` by `--revsion` to accept git SHA, branch names, or tags.
- Implemented checks to verify the existence of the specified git reference.
- Stored the initial git reference (branch or SHA) at the start of the script and restored it upon completion or exit, maintaining the initial state.

Closes #117

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -149,6 +149,15 @@ upgrade_venv(){
...
 }
 
+restore_git_ref() {
+    echo "Restoring to initial git reference: $initial_git_ref"
+    git checkout "$initial_git_ref" --quiet
+}
+
+# Ensure to recover current git branch/tag/SHA on script exit
+initial_git_ref=$(git rev-parse --abbrev-ref HEAD || git rev-parse HEAD)
+trap restore_git_ref EXIT
```
Comment (Shettland) [inline install.sh:159]: I did not know that "trap" existed, cool one!

File: install.sh [hunk 2]
```diff
@@ -375,10 +399,10 @@ if [ $upgrade == true ]; then
...
     # update installation by sinchronize folders
     echo "Copying files to installation folder"
-        rsync -rlv conf/ $INSTALL_PATH/conf/
-        rsync -rlv --fuzzy --delay-updates --delete-delay \
-              --exclude "logs" --exclude "documents" --exclude "migrations" --exclude "__pycache__" \
-              README.md LICENSE conf django_utils $REQUIRED_MODULES $INSTALL_PATH/
+    rsync -rlv conf/ $INSTALL_PATH/conf/
+    rsync -rlv --fuzzy --delay-updates --delete-delay \
+        --exclude "logs" --exclude "documents" --exclude "migrations" --exclude "__pycache__" \
+        README.md LICENSE conf django_utils $REQUIRED_MODULES $INSTALL_PATH/
```
Comment (Shettland) [inline install.sh]: Might remove django_utils from here aswell

File: install.sh (hunk 1 above)
Comment (Daniel-VM) [reply install.sh:159]: Same here, I just discovered

**Review by Shettland — APPROVED** [review]
LGTM! We should try to clean it more in the future, for example `conf/first_install_tables.json` file does not exist. But let's go step by step

**Comment by Daniel-VM** [comment]
Thanks @Shettland ! Yes, you are right. There are few remaining warnings too. We will address those in the near future. 
One more thing, the git CI tests haven't triggered 🤔

**Comment by Shettland** [comment]
> Thanks @Shettland ! Yes, you are right. There are few remaining warnings too. We will address those in the near future. One more thing, the git CI tests haven't triggered 🤔

Im trying to fix the same issue with the CI in https://github.com/BU-ISCIII/buisciii-tools, If it works, I will implement it here too

**Pull request merged**